base/expr.jl

```diff
@@ -9,7 +9,7 @@ const is_expr = isexpr
 """
     gensym([tag])
 
-Generates a symbol which will not conflict with other variable names.
+Generates a symbol which will not conflict with other variable names (in the same module).
```

Comment:
FWIW, with a package like:
```julia
module TestPkg
g = gensym()
f() = gensym() == g
end # module
```

calling `f` will return true. So this PR's wording isn't quite right? 

In any case, I don't  see a concise way of explaining #44903, I think it's fine to just close it.